float32 ref_vel
float32 cur_vel
